connectors/class-connector-users.php

```diff
@@ -387,4 +388,29 @@ public function callback_deleted_user( $user_id ) {
 			$user->ID
 		);
 	}
+
+	/**
+	 * Logs password-less user login
+	 *
+	 * @action set_current_user
+	 */
+	public function callback_set_current_user() {
+		$user_id = get_current_user_id();
+
+		if ( 0 === $user_id ) {
+			return;
+		}
+
+		$user = get_user_by( 'id', $user_id );
+
+		$this->log(
+			/* translators: %s: a user display name (e.g. "Jane Doe") */
+			__( '%s logged in', 'stream' ),
```

Comment:
@kidunot89 I would recommend changing the wording here, because `wp_set_current_user` can be used to achieve a variety of effects, not just to facilitate authentication via external services.

Maybe something like: `'%s set as the active user'`?

Reply:
Also, should we also respond to `wp_set_current_user( 0 )` – something like `'Active user unset'`?

Reply:
I don't think so because `wp_set_current_user( 0 )` is used to end the user session typically.

Reply:
@kidunot89 OK, fair enough.

What do you think about the other suggestion? If Stream is designed to capture administrative actions, should we even observe `set_current_user`, which is generally used programatically?

Reply:
Some plugins like OneLogin that rely `wp_set_current_user()` to authenticate to user. See this issue [here](https://github.com/xwp/stream/issues/946)

Reply:
> See this issue here.

I'm aware of the issue and the _intent_ here, @kidunot89 – what I'm saying is `wp_set_current_user` is used in many contexts and it is being used programmatically. So:

* If Stream is supposed to mainly track administrative actions, observing `set_current_users` will lead to tracking many programmatic actions instead and to reporting log ins where no log ins are actually happening.
* If we want to track these programmatic actions as well, we should track its opposite as well, i.e. `wp_set_current_user( 0 )`.

As for SSO plugins, we should probably deal with them on individual basis. OneLogin, for instance, uses this code:

```php
if (is_a($user_id, 'WP_Error')) {
	$errors = $user_id->get_error_messages();
	foreach($errors as $error) {
		echo esc_html($error).'<br>';
	}
	exit();
} else if ($user_id) {
	wp_set_current_user($user_id);
	
	$rememberme = false;
	$remembermeMapping = get_option('onelogin_saml_attr_mapping_rememberme');
	if (!empty($remembermeMapping) && isset($attrs[$remembermeMapping]) && !empty($attrs[$remembermeMapping][0])) {
		$rememberme = in_array($attrs[$remembermeMapping][0], array(1, true, '1', 'yes', 'on')) ? true : false;
	}
	wp_set_auth_cookie($user_id, $rememberme);

	setcookie(SAML_LOGIN_COOKIE, 1, time() + YEAR_IN_SECONDS, SITECOOKIEPATH );
	setcookie(SAML_NAMEID_COOKIE, $auth->getNameId(), time() + YEAR_IN_SECONDS, SITECOOKIEPATH );
	setcookie(SAML_SESSIONINDEX_COOKIE, $auth->getSessionIndex(), time() + YEAR_IN_SECONDS, SITECOOKIEPATH );
	setcookie(SAML_NAMEID_FORMAT_COOKIE, $auth->getNameIdFormat(), time() + YEAR_IN_SECONDS, SITECOOKIEPATH );
}
```

... which means that whenever `wp_set_current_user` is called, `wp_set_auth_cookie` is called as well, which in turn runs the `set_logged_in_cookie` action. There is apparently no reason why Stream would not track those log-ins.

Have you investigated why the cookie action is not run? Seeing that the issue is rather old, is it possible the plugin code changed in the meantime?

Reply:
Possibly 🤔

Reply:
@dero I believe you are right. This callback would add too much noise and unpredictability to the records.